backend/schema/schema.go

```diff
@@ -90,7 +93,8 @@ func (s *Schema) ResolveRefToType(ref *Ref, out Decl) error {
 			}
 		}
 	}
-	return fmt.Errorf("could not resolve reference %v", ref)
+
+	return errors.Join(ErrNotFound, fmt.Errorf("could not resolve reference %v", ref))
```

Comment:
This is great, but can you add this to the other `Resolve*` methods, and add a comment to each, and the error, mentioning that this is returned when the reference cannot be found?

Reply:
`ResolveRefMonomorphised` is the only one that seems relevant because it calls this